scopes/envs/envs/envs.docs.md

```diff
@@ -1,33 +1,628 @@
+    private tsAspect: TypescriptMain
+){}
+
+// ...
+
+getCompiler() {
+    const tsConfig = require.resolve('./typescript/tsconfig.json')
+    return this.tsAspect.createCompiler(tsConfig);
+}
 ```
 
-# Configure workspace to use the env
+#### getLinter
 
-add this to your workspace.jsonc file.
+```ts
+getLinter(...args : any[]): Linter
 ```
-"teambit.envs/envs": {
-  "env": "your-new-env",
-  "config": {}
+
+Returns a linter to be used by the Linter service.
+
+For example:
+
+```ts
+export class ReactEnv implements Environment {
+
+    constructor(){
+        // ...
+
+        // The ESLint aspect
+        private eslint: ESLintMain
+    }
+
+    // ...
+
+    getLinter() {
+        const eslintConfig = require.resolve('./eslint/eslintrc')
+        return this.eslint.createLinter({
+            config: eslintConfig,
+            // resolve all plugins from the react environment
+            pluginPath: __dirname,
+        });
+    }
+}
+```
+
+#### getDevServer
+
+```ts
+getDevServer(...args : any[]): DevServer
+```
+
+Returns a DevServer to be used by the DevServer service. (A DevServer is essentially the combination of the bundler configurations, together with a specified 'listen' port number)
+
+For example:
+
+```ts
+export class ReactEnv implements Environment {
+  constructor(
+    // ...
+
+    // The Webpack aspect
+    private webpack: WebpackMain
+  ) {}
+
+  // ...
+
+  getDevServer(): DevServer {
+    const withDocs = Object.assign(context, {
+      entry: context.entry.concat([require.resolve("./docs")]),
+    });
+    return this.webpack.createDevServer(withDocs, webpackConfig);
+  }
 }
 ```
 
-# Composing an Environment
-TBD
+> The above example runs the dev server with the environment's documentation template.
+
+#### getDocsTemplate
+
+```ts
+getDocsTemplate(...args : any[]): string
+```
+
+Returns the path to the documentation template files, to be used by the Documentation service.
+
+For example (see docs files [here](https://github.com/teambit/bit/tree/master/scopes/react/react/docs)):
+
+```ts
+export class ReactEnv implements Environment {
+  // ...
+
+  getDocsTemplate() {
+    return require.resolve("./docs");
+  }
+}
+```
+
+#### getPackageJsonProps
+
+```ts
+getPackageJsonProps(...args : any[]): object
+```
+
+Returns an object that defines the `package.json` properties of the packages generated for components handled by this environment. This configuration is used by the Packager service.
+
+Learn more about overriding the `package.json` properties [here](/docs/packages/publish-to-npm#packagejson)
+
+```ts
+export class ReactEnv implements Environment {
+  // ...
+
+  getPackageJsonProps() {
+    return {
+      main: "dist/{main}.js",
+      types: "{main}.ts",
+    };
+  }
+}
+```
+
+> As with any other 'merging' process, the properties defined in the above returned object will be added to the default configurations. Conflicting properties will be overridden by the properties set here.
```

Comment:
both getDepenednceis and getPacakgeJsonProps, has 4 sources that are merged:
1. the user configuration in the variant / component.json
2. a slot provided by the depResolver / pkg aspects.
3. the env - those function here
4. auto calculated by bit internals
This is the order (1 is strongest)